contracts/HybridCustody.cdc

```diff
@@ -70,49 +76,52 @@ pub contract HybridCustody {
     ///
-    pub resource interface BorrowableAccount {
-        access(contract) fun borrowAccount(): &AuthAccount
-        pub fun check(): Bool
+    access(all) resource interface BorrowableAccount {
+        access(contract) view fun _borrowAccount(): auth(Storage, Contracts, Keys, Inbox, Capabilities) &Account
+        access(all) view fun check(): Bool
     }
 
     /// Public methods anyone can call on an OwnedAccount
     ///
-    pub resource interface OwnedAccountPublic {
+    access(all) resource interface OwnedAccountPublic {
         /// Returns the addresses of all parent accounts
-        pub fun getParentAddresses(): [Address]
+        access(all) view fun getParentAddresses(): [Address]
 
         /// Returns associated parent addresses and their redeemed status - true if redeemed, false if pending
-        pub fun getParentStatuses(): {Address: Bool}
+        access(all) view fun getParentStatuses(): {Address: Bool}
 
         /// Returns true if the given address is a parent of this child and has redeemed it. Returns false if the given
         /// address is a parent of this child and has NOT redeemed it. Returns nil if the given address it not a parent
         /// of this child account.
-        pub fun getRedeemedStatus(addr: Address): Bool?
+        access(all) view fun getRedeemedStatus(addr: Address): Bool?
 
         /// A callback function to mark a parent as redeemed on the child account.
         access(contract) fun setRedeemed(_ addr: Address)
+
+        /// A helper function to find what controller Id to ask for if you are looking for a specific type of capability
+        access(all) view fun getControllerIDForType(type: Type, forPath: StoragePath): UInt64?
     }
 
     /// Private interface accessible to the owner of the OwnedAccount
     ///
-    pub resource interface OwnedAccountPrivate {
+    access(all) resource interface OwnedAccountPrivate {
         /// Deletes the ChildAccount resource being used to share access to this OwnedAccount with the supplied parent
         /// address, and unlinks the paths it was using to reach the underlying account.
-        pub fun removeParent(parent: Address): Bool
+        access(Owner | Remove) fun removeParent(parent: Address): Bool
 
         /// Sets up a new ChildAccount resource for the given parentAddress to redeem. This child account uses the
         /// supplied factory and filter to manage what can be obtained from the child account, and a new
         /// CapabilityDelegator resource is created for the sharing of one-off capabilities. Each of these pieces of
         /// access control are managed through the child account.
-        pub fun publishToParent(
+        access(Publish | Owner) fun publishToParent(
             parentAddress: Address,
-            factory: Capability<&CapabilityFactory.Manager{CapabilityFactory.Getter}>,
+            factory: Capability<&{CapabilityFactory.Getter}>,
```

Comment:
Since the capability was a restricted type before, doesn't that mean they will be migrated to `&CapabilityFactory.Manager` instead of `&{CapabilityFactory.Getter}`?  So should the type here be `Manager` instead?

Reply:
We should probably lean towards keeping things as open as possible unless there is a good reason not to

Reply:
Left a note on the PR itself, let's bring conversation about this topic there so we can bring all discussion about it to one place versus across comments